src/main/java/org/osiam/storage/query/UserFilterParser.java

```diff
@@ -50,4 +50,8 @@ public UserFilterParser(ExtensionDao extensionDao) {
         return UserQueryField.fromString(sortBy.toLowerCase(Locale.ENGLISH));
     }
 
+    @Override
+    protected UserSimpleFilterChain createFilterChain(FilterExpression<UserEntity> filter) {
+        return new UserSimpleFilterChain(entityManager.getCriteriaBuilder(), extensionDao, (UserFilterExpression) filter);
```

Comment:
Because it's a different method then. The abstract method needs any `FilterExpression<UserEntity>` not just instances of `UserFilterExpression`. I'll take a look at this.